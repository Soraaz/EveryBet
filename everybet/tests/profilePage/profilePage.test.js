import React from "react";
import {NavigationEvents} from "react-navigation";
import renderer from "react-test-renderer";
import {shallow} from "enzyme/build";

import ProfilePage from "../../src/profilePage/profilePage.js";
import User from "../../src/classes/user";
import LoginComponent from "../../src/profilePage/components/loginComponent";
import ProfileComponent from "../../src/profilePage/components/profileComponent";

describe("ProfilePage", () => {
	describe("rendering", () => {
		it("Should render correctly with LoginComponent", () => {
			User.id = undefined;
			const tree = renderer
				.create(<ProfilePage />)
				.toJSON();
			expect(tree).to.matchSnapshot();
		});

		it("Should render correctly with ProfileComponent", () => {
			User.id = 42;
			const tree = renderer
				.create(<ProfilePage />)
				.toJSON();
			expect(tree).to.matchSnapshot();
		});
	});

	describe("functionalities", () => {
		let wrapper;

		beforeEach(() => {
			User.id = undefined;
			wrapper = shallow(<ProfilePage />);
		});

		it("Should call the _onWillFocus callback", () => {
			const spy = sinon.spy(wrapper.instance(), "_onWillFocus");
			wrapper.find(NavigationEvents).first().props().onWillFocus();
			expect(spy).to.be.calledOnce;
		});

		it("Should change the child component if there's a defined user id", () => {
			expect(wrapper.find(LoginComponent).exists()).to.equal(true);
			expect(wrapper.find(ProfileComponent).exists()).to.equal(false);
			User.id = 42;
			wrapper.instance()._onWillFocus();
			expect(wrapper.find(LoginComponent).exists()).to.equal(false);
			expect(wrapper.find(ProfileComponent).exists()).to.equal(true);
		});
	});
});
