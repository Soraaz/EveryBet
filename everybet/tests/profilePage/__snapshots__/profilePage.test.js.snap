// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePage rendering Should render correctly with LoginComponent 1`] = `
<RCTSafeAreaView
  style={
    Array [
      Object {
        "flex": 1,
        "justifyContent": "flex-start",
      },
      Object {
        "backgroundColor": "#37597F",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "justifyContent": "flex-start",
        },
        Object {
          "backgroundColor": "#F4F4FE",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#37597F",
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "flex-start",
        }
      }
    >
      <Image
        accessible={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        source={
          Object {
            "testUri": "../../../assets/images/everybet_logo.png",
          }
        }
        style={
          Object {
            "height": 39.25,
            "marginLeft": 25,
            "width": 138,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 10,
            "flexDirection": "row",
            "height": 20,
            "justifyContent": "flex-start",
            "marginLeft": "auto",
            "width": 90,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../assets/images/betcoin_icon.png",
            }
          }
          style={
            Object {
              "height": 30,
              "width": 30,
            }
          }
        />
        <Text
          style={
            Object {
              "fontFamily": "System",
              "fontSize": 10,
              "marginLeft": "auto",
              "marginRight": 5,
              "textAlign": "right",
            }
          }
        >
          0
        </Text>
      </View>
      <Image
        accessible={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        source={
          Object {
            "testUri": "../../../assets/images/user_placeholder.png",
          }
        }
        style={
          Object {
            "borderRadius": 20,
            "height": 40,
            "marginHorizontal": 15,
            "width": 40,
          }
        }
      />
    </View>
    <Text
      style={
        Object {
          "fontFamily": "System",
          "fontSize": 20,
          "fontWeight": "600",
          "marginTop": 25,
          "textAlign": "center",
          "textAlignVertical": "center",
        }
      }
    >
      Connectez-vous pour commencer Ã  parier !
    </Text>
    <View
      style={
        Object {
          "marginTop": 20,
        }
      }
    >
      <View
        style={Object {}}
      >
        <View
          style={
            Object {
              "marginBottom": 10,
            }
          }
        >
          <View>
            <TextInput
              allowFontScaling={true}
              onChangeText={[Function]}
              placeholder="Identifiant"
              placeholderTextColor="#DBDBDB"
              style={
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "#FFFFFF",
                  "borderRadius": 3,
                  "fontFamily": "System",
                  "fontSize": 15,
                  "height": 50,
                  "marginTop": 20,
                  "paddingLeft": 20,
                  "shadowColor": "#000000",
                  "shadowOffset": Object {
                    "height": 3,
                    "width": 0,
                  },
                  "shadowOpacity": 0.33,
                  "shadowRadius": 3,
                  "width": 282,
                }
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </View>
        <View
          style={
            Object {
              "marginBottom": 10,
            }
          }
        >
          <View>
            <TextInput
              allowFontScaling={true}
              onChangeText={[Function]}
              placeholder="Mot de passe"
              placeholderTextColor="#DBDBDB"
              secureTextEntry={true}
              style={
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "#FFFFFF",
                  "borderRadius": 3,
                  "fontFamily": "System",
                  "fontSize": 15,
                  "height": 50,
                  "marginTop": 20,
                  "paddingLeft": 20,
                  "shadowColor": "#000000",
                  "shadowOffset": Object {
                    "height": 3,
                    "width": 0,
                  },
                  "shadowOpacity": 0.33,
                  "shadowRadius": 3,
                  "width": 282,
                }
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "column",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "marginRight": 30,
                "marginTop": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "alignItems": "stretch",
                  "flex": 4,
                  "fontFamily": "System",
                  "fontSize": 15,
                  "textAlign": "center",
                }
              }
            >
              Se souvenir de moi
            </Text>
            <RCTSwitch
              accessibilityLabel={null}
              onChange={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "height": 31,
                    "width": 51,
                  },
                  Array [
                    Object {
                      "alignSelf": "flex-end",
                    },
                    Object {
                      "backgroundColor": "#989898",
                      "borderRadius": 16,
                    },
                  ],
                ]
              }
              thumbTintColor="#F1F1F1"
              tintColor="#989898"
              value={false}
            />
          </View>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#A6E48D",
            "borderRadius": 10,
            "height": 45,
            "justifyContent": "center",
            "marginTop": 20,
            "opacity": 1,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 6,
              "width": 0,
            },
            "shadowOpacity": 0.33,
            "shadowRadius": 7.5,
            "width": 215,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 15,
              "textAlign": "center",
            }
          }
        >
          Se connecter
        </Text>
      </View>
    </View>
    <Text
      style={
        Object {
          "fontFamily": "System",
          "fontSize": 20,
          "fontWeight": "600",
          "marginTop": 25,
          "textAlign": "center",
          "textAlignVertical": "center",
        }
      }
    >
      Pas encore inscrit ?
    </Text>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 10,
          "height": 45,
          "justifyContent": "center",
          "marginTop": 20,
          "opacity": 1,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 6,
            "width": 0,
          },
          "shadowOpacity": 0.33,
          "shadowRadius": 7.5,
          "width": 215,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontSize": 15,
            "textAlign": "center",
          }
        }
      >
        S'inscrire
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`ProfilePage rendering Should render correctly with ProfileComponent 1`] = `
<RCTSafeAreaView
  style={
    Array [
      Object {
        "flex": 1,
        "justifyContent": "flex-start",
      },
      Object {
        "backgroundColor": "#37597F",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
        },
        Object {
          "backgroundColor": "#F4F4FE",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "flex-start",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/images/user_placeholder.png",
          }
        }
        style={
          Object {
            "alignSelf": "center",
            "borderRadius": 65,
            "height": 125,
            "width": 125,
          }
        }
      />
      <Text
        style={
          Object {
            "fontFamily": "System",
            "fontSize": 32,
            "fontWeight": "600",
            "textAlign": "center",
            "textAlignVertical": "center",
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "marginBottom": 10,
            "marginLeft": 45,
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "fontFamily": "System",
                "fontSize": 20,
                "fontWeight": "600",
                "marginTop": 25,
                "textAlign": "center",
                "textAlignVertical": "center",
              }
            }
          >
            BetCoins
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderRadius": 10,
                "flexDirection": "row",
                "height": 20,
                "justifyContent": "flex-start",
                "width": 90,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontFamily": "System",
                  "fontSize": 10,
                  "marginLeft": "auto",
                  "marginRight": 5,
                  "textAlign": "right",
                }
              }
            />
          </View>
        </View>
        <Image
          source={
            Object {
              "testUri": "../../../assets/images/betcoin_icon.png",
            }
          }
          style={
            Object {
              "height": 40,
              "marginLeft": 5,
              "marginTop": 30,
              "width": 40,
            }
          }
        />
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 10,
            "height": 45,
            "justifyContent": "center",
            "marginTop": 20,
            "opacity": 1,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 6,
              "width": 0,
            },
            "shadowOpacity": 0.33,
            "shadowRadius": 7.5,
            "width": 215,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 15,
              "textAlign": "center",
            }
          }
        >
          Editer le profil
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 10,
            "height": 45,
            "justifyContent": "center",
            "marginTop": 20,
            "opacity": 1,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 6,
              "width": 0,
            },
            "shadowOpacity": 0.33,
            "shadowRadius": 7.5,
            "width": 215,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 15,
              "textAlign": "center",
            }
          }
        >
          Proposer un pari
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 10,
            "height": 45,
            "justifyContent": "center",
            "marginTop": 20,
            "opacity": 1,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 6,
              "width": 0,
            },
            "shadowOpacity": 0.33,
            "shadowRadius": 7.5,
            "width": 215,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 15,
              "textAlign": "center",
            }
          }
        >
          Paris en cours
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 10,
            "height": 45,
            "justifyContent": "center",
            "marginTop": 20,
            "opacity": 1,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 6,
              "width": 0,
            },
            "shadowOpacity": 0.33,
            "shadowRadius": 7.5,
            "width": 215,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 15,
              "textAlign": "center",
            }
          }
        >
          Historique des paris
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 10,
            "height": 45,
            "justifyContent": "center",
            "marginTop": 20,
            "opacity": 1,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 6,
              "width": 0,
            },
            "shadowOpacity": 0.33,
            "shadowRadius": 7.5,
            "width": 215,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 15,
              "textAlign": "center",
            }
          }
        >
          Regarder une pub pour gagner des piÃ¨ces
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 10,
            "height": 45,
            "justifyContent": "center",
            "marginTop": 20,
            "opacity": 1,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 6,
              "width": 0,
            },
            "shadowOpacity": 0.33,
            "shadowRadius": 7.5,
            "width": 215,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 15,
              "textAlign": "center",
            }
          }
        >
          Contacter Everybet
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#FC8D8D",
            "borderRadius": 10,
            "height": 45,
            "justifyContent": "center",
            "marginTop": 20,
            "opacity": 1,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 6,
              "width": 0,
            },
            "shadowOpacity": 0.33,
            "shadowRadius": 7.5,
            "width": 215,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 15,
              "textAlign": "center",
            }
          }
        >
          Se dÃ©connecter
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
